### ISSUE TITLE
compiling on ubuntu 23:10: libcaesium.so, remove cargo, sources/build files, qt - sudo make install, then remove all sources

### ISSUE THREAD

**Issue by JeckPeppinger** [Lymphatus/caesium-image-compressor#269]
First of all - compiling on ubuntu 23.10 worked, and CIC works. Thx!!

pls. can you help with the command to install the CIC binary and the so's (and any other runtime dependencies if any)?

I tried sudo cmake install but cannot get the right command.

I want to install CiC and all runtime dependencies for all users on the machine, and then remove qt, sources, cargo to save space.

Thanks for any hint!

**Comment by JeckPeppinger** [comment]
After building again with the cmake commands as given in the readme.md the following error arises:
caesium-image-compressor: error while loading shared libraries: libQt6Concurrent.so.6: cannot open shared object file: No such file or directory
 
This error came just after the second compile!

**Comment by Lymphatus** [comment]
The process on Linux is a bit more complicated than on other systems. If you want to compile and then remove all of build dependencies, you can use tool like [linuxdeployqt](https://github.com/probonopd/linuxdeployqt), which will make a "self-contained" application. You can also take a look at #259 which is related to this issue

**Comment by JeckPeppinger** [comment]
"caesium-image-compressor: error while loading shared libraries: libQt6Concurrent.so.6: cannot open shared object file: No such file or directory"
Any hint how to resolve that?

I am really surprised that linux is not better supported.
download, build, install, cleanup - long since i programmed myself, but this was a standard.
The method you described for building on linux costs at least 5GB on the disk. A bit perverted to have a 5GB tool that helps reduce filesizes of pictures ....

Can you describe please how to install properly without need for carrying 5GB on a disk for the sources?

**Comment by Lymphatus** [comment]
TL;DR
You can use this [third-party build](https://github.com/larygwil/caesium-image-compressor/releases) that ships an AppImage. But I'm not sure if it runs on Ubuntu 23.10.

Caesium is a UI software based on Qt6. If you want to build it, you need to install Qt, which is very large indeed. This is only needed for _compiling_ it, not running it. Once you compile it, the actual executable needs to have some libraries still on the system, one of them is `libQt6Concurrent.so.6`, but there are a few more.
The process would roughly be:
- Install Qt6 dev packages (very large)
- Compile the software (cmake, make)

At this point if you keep the Qt libraries, it should work. If you want to cleanup all the compilation requirements, you need to manually (or use [linuxdeployqt](https://github.com/probonopd/linuxdeployqt)) keep/copy the required libraries in a place where the executable can find them (the same folder works), because they are dynamically linked, not statically linked. Once you've done that, you can unistall Qt6 and all dependencies.
This is the same way that me or you would do on Windows or macOS. The difference is that on Windows I can do this process myself and give you an installer or a zip* with an .exe, and on macOS I can do the same and give a .app, which is a self contained folder with all requirements. Not on Linux, because a binary build on Ubuntu might not run on Fedora, or Suse or any other distro. 

In #259 we are discussing a way to simplify the Linux binary delivery.
